lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -602,26 +602,51 @@ def write(chunk)
 
           rescue GRPC::BadStatus => error
             increment_failed_requests_count(error.code)
+            # See the mapping between http status and grpc status code at:
+            # https://github.com/grpc/grpc/blob/master/src/core/lib/transport/status_conversion.cc
             case error.code
-            when GRPC::Core::StatusCodes::CANCELLED,
-                 GRPC::Core::StatusCodes::UNAVAILABLE,
-                 GRPC::Core::StatusCodes::DEADLINE_EXCEEDED,
-                 GRPC::Core::StatusCodes::INTERNAL,
-                 GRPC::Core::StatusCodes::UNKNOWN
-              # Server error, so retry via re-raising the error.
+            # Server error, so retry via re-raising the error.
+            when \
+                # Maps to http status 500 (Internal Server Error)
+                GRPC::Core::StatusCodes::INTERNAL,
+                # Maps to http status 501 (Not Implemented).
+                GRPC::Core::StatusCodes::UNIMPLEMENTED,
+                # Maps to http status 503 (Service Unavailable).
+                GRPC::Core::StatusCodes::UNAVAILABLE,
+                # Maps to http status 504 (Gateway Timeout).
+                GRPC::Core::StatusCodes::DEADLINE_EXCEEDED
               increment_retried_entries_count(entries_count, error.code)
               @log.debug "Retrying #{entries_count} log message(s) later.",
                          error: error.to_s, error_code: error.code.to_s
               raise error
+
+            # Most client errors indicate a problem with the request itself and
+            # should not be retried.
             when \
-                # Most client errors indicate a problem with the request itself
-                # and should not be retried.
-                GRPC::Core::StatusCodes::UNIMPLEMENTED,
+                # Maps to http status 400 (Bad Request).
+                GRPC::Core::StatusCodes::INVALID_ARGUMENT,
+                # Authorization error. These are usually solved via a `gcloud
```

Comment:
Let's reorder these:
```ruby
                # Authorization error. Maps to HTTP status 401 (Unauthorized).
                # These are usually solved via a `gcloud auth` call, or by
                # modifying the permissions on the Google Cloud project.
```

Reply:
Changed to:

```
                # HTTP status 401 (Unauthorized).
                # These are usually solved via a `gcloud auth` call, or by
                # modifying the permissions on the Google Cloud project.
```

The `Authorization error.` part seems redundant.